src/mapd-con-es6.js

```diff
@@ -294,45 +299,97 @@ class MapdCon {
     this.queryTimes[queryId] = execution_time_ms
   }
 
-  promisifyThriftMethod = (client, sessionId, methodName, args) =>
+  // ** Event publishing **
+
+  eventListeners = {
+    error: []
+  }
+
+  /**
+   * Subscribe a listener to be called when a given event occurs.
+   * @param {String} eventName - An event name - must be a key of eventListeners.
+   * @param {String} callback - Listener function to be called.
+   * @returns {Function} A function to unsubscribe this listener.
+   *
+   * @example <caption>Get a list of all users:</caption>
+   *
+   * con.getUsersAsync().then(res => console.log(res))
+   */
+  on = (eventName, callback) => {
+    if (!this.eventListeners[eventName]) {
+      throw new Error(`Invalid event name: ${eventName}`)
+    }
+
+    if (typeof callback !== "function") {
+      throw new Error("Event listener must be a function")
+    }
+
+    this.eventListeners[eventName].push(callback)
+
+    // Unsubscribe callback
+    return () => {
+      this.eventListeners[eventName] = this.eventListeners[eventName].filter(
+        listener => listener !== callback
+      )
+    }
+  }
+
+  publish = (eventName, payload) => {
+    // Don't block on listeners to complete
+    setTimeout(() => {
+      this.eventListeners[eventName].forEach(listener => {
```

Comment:
We should have a check that the `eventName` exists, so that we're not calling `forEach` on a null.

Reply:
Outdated - removed and replaced with eventemitter3